File: haskell/src/LambdaGardenDue.hs
module LambdaGardenDue(
  aggrega, vincitoreRiga,
  cellaSuccessivaRighe, cellaSuccessivaColonne, cellaSuccessivaDiagonaleMaggiore, cellaSuccessivaDiagonaleMinore,
  generaVettore, vincitoreMatrice
) where

import Data.List --Serve per transpose

aggrega :: Char -> Char -> Char
aggrega x y = 'A'

vincitoreRiga :: [Char] -> Char
vincitoreRiga xs = 'A'

cellaSuccessivaRighe :: (Int, Int) -> (Int, Int)
cellaSuccessivaRighe (x,y) = (-1,-1)

cellaSuccessivaColonne :: (Int, Int) -> (Int, Int)
cellaSuccessivaColonne (x,y) = (-1,-1)

cellaSuccessivaDiagonaleMaggiore :: (Int, Int) -> (Int, Int)
cellaSuccessivaDiagonaleMaggiore (x,y) = (-1,-1)

cellaSuccessivaDiagonaleMinore :: (Int, Int) -> (Int, Int)
cellaSuccessivaDiagonaleMinore (x,y) = (-1,-1)

generaVettore :: [[Char]] -> (Int,Int) -> ((Int, Int) -> (Int, Int)) -> [Char]
generaVettore matrice (x,y) funSuccessore = []

aggregaVincitore :: Char -> Char -> Char
aggregaVincitore x y = 'A'

trovaVincitoreRighe :: [[Char]] -> Char
trovaVincitoreRighe matrice = 'A'

trovaVincitoreColonne :: [[Char]] -> Char
trovaVincitoreColonne matrice = 'A'

vincitoreMatrice :: [[Char]] -> Char
vincitoreMatrice matrice = 'A'

File: java/src/main/java/lg2/LambdaGardenDue.java
package lg2;

import java.util.Arrays;
import java.util.Optional;
import java.util.function.UnaryOperator;
import java.util.stream.Stream;

public class LambdaGardenDue {

    static class Cella{
        public Cella(int x, int y){
            this.x=x;
            this.y=y;
        }
        public int x,y;

        @Override
        public boolean equals(Object obj) {
            if(!(obj instanceof Cella))return false;
            Cella altra = (Cella)obj;
            return this.x==altra.x && this.y==altra.y;
        }

        @Override
        public String toString() {
            return "x: " + x + ", y: " + y;
        }
    }

    static final Cella CELLA_ZERO = new Cella(0,0);

    public Character aggrega(Character a, Character b){
        return 'A';
    }

    public char vincitore(Character[] riga){
        return 'A';
    }

    public Cella cellaSuccessivaRighe(Cella c){
        return new Cella(-1,-1);
    }

    public Cella cellaSuccessivaColonne(Cella c){
        return new Cella(-1,-1);
    }

    public Cella cellaSuccessivaDiagonaleMaggiore(Cella c){
        return new Cella(-1,-1);
    }

    public Cella cellaSuccessivaDiagonaleMinore(Cella c){
        return new Cella(-1,-1);
    }

    Optional<Character> estrai(Character[][] partita, Cella cella){
        return Optional.empty();
    }


    public Character[] generaVettore(Character[][] partita, Cella inizio, UnaryOperator<Cella> successore){
        return new Character['A'];
    }

    public Stream<Character> generaStream(Character[][] partita, Cella inizio, UnaryOperator<Cella> successore){
        return Stream.empty();
    }

    public Character aggregaVincitore(Character a, Character b){
        return 'A';
    }

    public Character trovaVincitoreRighe(Character[][] partita){
        return 'A';
    }

    public Character trovaVincitoreColonne(Character[][] partita){
        return 'A';
    }


    public Character trovaVincitoreDiagonaleMaggiore(Character[][] partita){
        return 'A';
    }

    public Character trovaVincitoreDiagonaleMinore(Character[][] partita){
        return 'A';
    }

    public Character vincitoreMatrice(Character[][] partita){
        return 'A';
    }
}